### PR TITLE
[CBRD-22910] heap_attrinfo_transform_to_disk resize record

### PR DESCRIPTION
**Pull request by ribram**
http://jira.cubrid.org/browse/CBRD-22910

### heap_attrinfo_transform_to_disk

1. change new_recdes type from RECDES * to record_descriptor *
2. when record data buffer is not enough, extend it by DB_PAGESIZE and retry
3. update serial_update_serial_object and locator_allocate_copy_area_by_attr_info to use new signature

### locator_allocate_copy_area_by_attr_info

1. remove while loop since heap_attrinfo_transform_to_disk should success on first try
2. consume record_descriptor allocated buffer; it there is an allocated buffer, it means that the initial copy area was not enough; release it and allocate another copy area big enough to hold all record data.

### record_descriptor

1. new setter function for recdes info: set_record_length, set_type
2. buffer manipulation functions: resize_buffer, set_external_buffer, release_buffer.

### CODE REVIEW

File: src/storage/heap_file.c [hunk 1]
```diff
@@ -11588,8 +11596,9 @@ heap_attrinfo_transform_to_disk_internal (THREAD_ENTRY * thread_p, HEAP_CACHE_AT
 
       if (ptr_varvals + mvcc_wasted_space >= buf->endptr)
 	{
-	  new_recdes->length = -expected_size;	/* set to negative */
-	  return S_DOESNT_FIT;
+	  // is it possible?
```
Comment (razvanradulescu) [inline src/storage/heap_file.c]: I think we should add 
assert(false);
here  and bellow on ER_TF_BUFFER_OVERFLOW
We should try to fix unmatched get_packed_size vs pack

**Comment by ribram** [comment]
We should run a manual test.

File: src/storage/heap_file.c (hunk 1 above)
Comment (ribram) [reply src/storage/heap_file.c]: Tests show that estimated size is imprecise and we'd probably have many crashes. We may consider this task for later.

File: src/query/serial.c [hunk 2]
```diff
@@ -832,9 +833,8 @@ static int
 serial_update_serial_object (THREAD_ENTRY * thread_p, PAGE_PTR pgptr, RECDES * recdesc, HEAP_CACHE_ATTRINFO * attr_info,
 			     const OID * serial_class_oidp, const OID * serial_oidp, DB_VALUE * key_val)
 {
-  RECDES new_recdesc;
-  char copyarea_buf[IO_MAX_PAGE_SIZE + MAX_ALIGNMENT];
-  int new_copyarea_length;
+  cubmem::stack_block < IO_MAX_PAGE_SIZE + MAX_ALIGNMENT > copyarea;
```
Comment (eseokoh) [inline src/query/serial.c]: Hope manual indentation :)

Comment (ribram) [reply src/query/serial.c]: added manual indentation :).

**Pull request merged**